pandas/core/accessor.py

```diff
@@ -58,7 +58,9 @@ def _delegate_method(self, name, *args, **kwargs):
         raise TypeError("You cannot call method {name}".format(name=name))
 
     @classmethod
-    def _add_delegate_accessors(cls, delegate, accessors, typ, overwrite=False):
+    def _add_delegate_accessors(
+        cls, delegate, accessors: List[str], typ: str, overwrite: bool = False
```

Comment:
the type annotation of `accessors` here needs to be compatible with `delegate_names` below.

`_add_delegate_accessors` is used in the inner function of `delegate_names`. This inner function `add_delegate_accessors` does not have type annotations and is therefore not checked by mypy. Annotating the inner functions gives `pandas\core\accessor.py:143: error: Argument 2 to "_add_delegate_accessors" of "PandasDelegate" has incompatible type "Union[List[str], Set[str]]"; expected "List[str]"`

Probably `Sequence[str]` is more appropriate, however beware a string is also a sequence of str.

```
from typing import Sequence


def foo(bar: Sequence[str]):
    pass


foo(3)
foo("hdfjjdhf")
foo([1, 2, 3])
foo(["bar", "baz"])
```
```
test.py:8: error: Argument 1 to "foo" has incompatible type "int"; expected "Sequence[str]"
test.py:10: error: List item 0 has incompatible type "int"; expected "str"
test.py:10: error: List item 1 has incompatible type "int"; expected "str"
test.py:10: error: List item 2 has incompatible type "int"; expected "str"
```

Reply:
Every combination I tried I got an error. I think I will drop the annotations for ```accessors``` in this PR.